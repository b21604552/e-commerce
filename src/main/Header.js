import React from 'react'
import Basket from '../partials/header/Basket'
import SearchBox from '../partials/header/SearchBox'
import InfoBar from '../partials/header/InfoBar'

class Header extends React.Component {

  constructor (props) {
    super(props)
    this.state = {}
  }

  render () {

    return (
      <>
        <div className={'header-wrapper'}>
          <div className={'header'}>
            <div className={'logo-wrapper'}/>
            <SearchBox/>
            <Basket visible={false}/>
          </div>
        </div>
        <InfoBar/>
      </>
    )
  }

}

export default Header