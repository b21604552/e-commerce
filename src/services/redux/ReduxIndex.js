import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
