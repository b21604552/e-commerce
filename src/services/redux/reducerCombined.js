import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
