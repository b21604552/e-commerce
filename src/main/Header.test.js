import React from 'react'
import ReactDOM from 'react-dom'
import Header from './Header'
import renderer from 'react-test-renderer'
import { reduxStore } from '../services/redux/ReduxIndex'
import { Provider } from 'react-redux'

describe('Header Test', () => {
  let page = (
    <Provider store={reduxStore}>
      <Header/>
    </Provider>
  )

  it('Renders Header without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(page, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders Header correctly', () => {
    const div = renderer.create(page).toJSON()
    expect(div).toMatchSnapshot()
  })
})