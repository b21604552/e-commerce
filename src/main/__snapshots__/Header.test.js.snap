// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Renders Header correctly 1`] = `
Array [
  <div
    className="header-wrapper"
  >
    <div
      className="header"
    >
      <div
        className="logo-wrapper"
      >
        <img
          alt="logo"
          src="logo.svg"
        />
      </div>
      <div
        className="search-wrapper"
      >
        <span
          className="prefix"
        >
          <img
            alt="search-icon"
            src="search-icon.svg"
          />
        </span>
        <input
          className="search-input"
          id="search-input"
          onChange={[Function]}
          placeholder="25 milyon’dan fazla ürün içerisinde ara"
          type="text"
        />
      </div>
      <div
        className="basket-wrapper"
      >
        <div
          className="basket-button"
        >
          <div>
            Sepetim
          </div>
        </div>
        <div
          className="basket-opened"
        >
          <div
            className="basket-empty-wrapper"
          >
            <div
              className="basket-empty"
            >
              Sepetinizde ürün bulunmamaktadır.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="info-wrapper"
  >
    <div
      className="category-wrapper"
    >
      <div
        className="category"
      >
        Hoşgeldiniz
      </div>
      <div
        className="search-keyword-wrapper"
      />
    </div>
    <div
      className="filter-wrapper"
    >
      <select
        className="custom-select"
        onChange={[Function]}
        value=""
      >
        <option
          disabled={true}
          hidden={true}
          value=""
        >
          Sıralama
        </option>
        <option
          value="0"
        >
          En Düşük Fiyat
        </option>
        <option
          value="1"
        >
          En Yüksek Fiyat
        </option>
        <option
          value="2"
        >
          En Yeniler (A&gt;Z)
        </option>
        <option
          value="3"
        >
          En Yeniler (Z&gt;A)
        </option>
      </select>
    </div>
  </div>,
]
`;
